### PR TITLE
esp32: esp8266: Rename dhcp_hostname to hostname

### PR DESCRIPTION
**Pull request by IhorNehrutsa**
This resolve misunderstanding from #8792
IRL dhcp_hostname is station or access point name.

### CODE REVIEW

File: docs/library/network.WLAN.rst [hunk 1]
```diff
@@ -130,7 +130,7 @@ Methods
    hidden         Whether SSID is hidden (boolean)
    security       Security protocol supported (enumeration, see module constants)
    key            Access key (string)
-   dhcp_hostname  The DHCP hostname to use
+   hostname       The station(access point) hostname to use
```
Comment (dpgeorge) [inline docs/library/network.WLAN.rst:133]: maybe make it clear that it sets both, eg "The hostname for the station and access point interfaces"

File: ports/esp32/network_wlan.c [hunk 2]
```diff
@@ -452,7 +452,7 @@ STATIC mp_obj_t network_wlan_config(size_t n_args, const mp_obj_t *args, mp_map_
                         cfg.ap.channel = mp_obj_get_int(kwargs->table[i].value);
                         break;
                     }
-                    case MP_QSTR_dhcp_hostname: {
+                    case MP_QSTR_hostname: {
```
Comment (dpgeorge) [inline ports/esp32/network_wlan.c:455]: I think we should provide some amount of backwards compatibility, at least for a few releases.  That's easy, just use:
```c
case MP_QSTR_hostname:
case MP_QSTR_dhcp_hostname : {
...
```

**Comment by IhorNehrutsa** [comment]
Ready

**Comment by dpgeorge** [comment]
Thanks for updating.  Merged in 1ea82b6dcc7dd5f0ad051db414ff91d8f04d20e0 with further tweaks to the docs entry.

**Pull request closed without merging**